plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XStatefulIncrementalBuilder.java

```diff
@@ -156,6 +156,10 @@ public XBuildResult launch() {
 
 		} catch (CancellationException e) {
 			// catch CancellationException here and proceed normally to save already resolved deltas
+		} catch (Throwable th) {
+			if (!operationCanceledManager.isOperationCanceledException(th)) {
+				throw th;
```

Comment:
How about
```
if (operationCanceledManager.isOperationCanceledException(th)) {
	// catch OperationCanceledException here and proceed normally to save already resolved deltas
} else {
	throw th;
}
```

Also: Why catching a `Throwable`? Above it is also an `Exception`

Reply:
Regarding "how about": actually I thought about that too, but then didn't like the duplication of the comment. But if you also stumbled across the readability here, let's use the more readable variant you proposed.

Regarding Throwable: method `#isOperationCanceledException()` accepts a throwable, i.e. may return true for throwables that are not exceptions; this is the case, for example, if a `OperationCanceledError` is thrown; so to cover all possible cases a throwable should be used.

However, I now adjusted the corresponding location in XBuildManager (changed from catching Exception to catching Throwable) for consistency.